Fan Param Actions refactor

silverailscolo opened the pull request:
- Use action targets instead of `device_id` field to follow HA (domain hvac + remote) > works
- Adjust SVC schemas > done

TODO:
- @wimpie70 confirm refactor with hardware and front-end

<img width="710" height="670" alt="FanParamsActionPaneNew" src="https://github.com/user-attachments/assets/d1f67fef-d905-4e86-b0eb-587782db06dd" />

silverailscolo commented:
This is mostly done, see my remarks above.
Fine to keep your methods in broker, accessible to both climate and remote.
Biggest change: entity service calls provide separate params, not a ServiceCall/dict object, so I  add the device_id and fold that into a dict before passing the call to broker. (simplified that too)

Didn't have to touch `ramses_rf`, but there was a device_id in the form "remote.12:345678" that somehow popped up in the _cc broker get_fan_param. Split it to pass. Use address?

I don't have a fan that wants to cooperate at startup.
Please have a good look and also check how you want to access this part from your .js add on.

silverailscolo commented on custom_components/ramses_cc/climate.py:
```diff
@@ -606,6 +606,58 @@ def preset_mode(self) -> str | None:
+        :type from_id: str
+        """
+        call: dict[str, Any] = {
+            "device_id": self.entity_id,
```
should be: `self.device_id`

wimpie70 commented:
What I found so far, (running out of time now, but i will continue and create proper commits):
There might be automations already depending on the servicecalls, Not sure if they will still work with the new format. Didn't test that yet. 

broker:
from_id for 2411 RQ/I messages can be a HGI, but when setting (W) a value, with HGI as a source, the FAN won't respond. It must come from device that was bound by the FAN. (defined in config_flow or not). HGI fallback was removed in #346.
I noticed some comments/user feedback still mentions HGI as an option. I'll try to find them all.

get/set_fan_param: fan_id, good catch, that's obsolete code from way back and is obsolete.

wimpie70 commented:
I think i have it working now, but was called afk. Will look at it again after diner

wimpie70 commented:
Why target a REM, Not the FAN ?
When i started on the 2411 params i 1st thought the REM should hold the logic for it, since real devices like a display do that too. but for ha thats not handy at all. The FAN is the device holding the real values. Keep it in 1 place, we can have multiple rem's.
However, with a co2 rem or other measuring rems this could be different for their entities since these can and will depend on their specific locations.

But it could be done. I would itterate over the devices to find a hvacVentilator and check that for the bound REM, if it has any. But maybe only on the fly, not build and store that dict and keep it updated. Depends on how often we expect it to be used i guess.


⁣BlueMail voor Android downloaden ​

Op 18 nov 2025, 22:57, om 22:57, Egbert Broerse ***@***.***> schreef:
>silverailscolo left a comment (ramses-rf/ramses_cc#350)
>
>@wimpie70 I disabled 2 tests that still used the "fan_id" field in the
>call data.
>
>Now, if I select a Remote as target in the Action UI, I would like to
>have the code find the Hvac device it is bound to. I think I can
>achieve that by building a dict of REMs in the broker, when init'ing
>the Hvac devices. Either the Remote service call or the
>Broker`_get_device_and_from_id` function could then swap the target for
>the from_id. You might already have given this some thought.
>BTW How did you apply the REM as a Target in the UI?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/ramses-rf/ramses_cc/pull/350#issuecomment-3549592912
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

silverailscolo commented:
@wimpie70 Can you verify this works with your plugin scripts? Got this working by making sure:
- the name of climate/remote UI services is not the same as the method on `broker.py`, else it would pickup the request and stumble on the call data.
- split get and set_param for remotes, looking up the "parent" bound device
- updated the translations to match
- not publishing extra services in init, or they fill the Action dropdown

Hope to hear
